File: 67-二进制求值.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int i = a.size() - 1, j = b.size() - 1, flag = 0;
        int a_val, b_val, total_val;
        string result;

        while(i >= 0 || j >= 0 || flag > 0){
            a_val = i >= 0 ? a[i--] - '0' : 0;
            b_val = j >= 0 ? b[j--] - '0' : 0;
            total_val = a_val + b_val + flag;
            if(total_val >= 2){
                flag = 1;
                result += (char)(total_val - 2 + '0');
            }
            else{
                flag = 0;
                result += (char)(total_val + '0');
            }
        }

        i = -1;
        j = result.size();
        char t;
        while(++i < --j){
            t = result[i];
            result[i] = result[j];
            result[j] = t;
        }

        return result;
    }
};

File: 67-二进制求值.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        c = []
        flag = 0
        i = len(a) - 1
        j = len(b) - 1
        
        while i >= 0 or j >= 0 or flag > 0:
            a_value = int(a[i]) if i >= 0 else 0
            b_value = int(b[j]) if j >= 0 else 0
            total_value = a_value + b_value + flag
            if total_value > 1:
                flag = 1
                c.append(str(total_value - 2))
            else:
                flag = 0
                c.append(str(total_value))
            i -= 1
            j -= 1
            
        return ''.join(reversed(c))
